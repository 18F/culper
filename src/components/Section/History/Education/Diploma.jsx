import React from 'react'
import { i18n } from '../../../../config'
import { ValidationElement, DateControl, Text, RadioGroup, Radio, Field, Show } from '../../../Form'

export class DiplomaItem extends ValidationElement {
  constructor (props) {
    super(props)

    this.update = this.update.bind(this)
    this.updateDiploma = this.updateDiploma.bind(this)
    this.updateDiplomaOther = this.updateDiplomaOther.bind(this)
    this.updateDate = this.updateDate.bind(this)
  }

  update (queue) {
    this.props.onUpdate({
      Diploma: this.props.Diploma,
      DiplomaOther: this.props.DiplomaOther,
      Date: this.props.Date,
      ...queue
    })
  }

  updateDiploma (values) {
    this.update({
      Diploma: values
    })
  }

  updateDiplomaOther (values) {
    this.update({
      DiplomaOther: values
    })
  }

  updateDate (values) {
    this.update({
      Date: values
    })
  }

  render () {
    return (
      <div>
        <Field title={i18n.t('history.education.heading.diploma')}
               help="history.education.help.diploma"
               adjustFor="big-buttons"
               shrink={true}
               scrollIntoView={this.props.scrollIntoView}>
          <RadioGroup className="diploma option-list"
                      required={this.props.required}
                      onError={this.props.onError}
                      selectedValue={(this.props.Diploma || {}).value}>
            <Radio name="diploma-highschool"
                   className="diploma-highschool"
                   label={i18n.m('history.education.label.diploma.highschool')}
                   value="High School Diploma"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-associate"
                   className="diploma-associate"
                   label={i18n.m('history.education.label.diploma.associate')}
                   value="Associate"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-bachelor"
                   className="diploma-bachelor"
                   label={i18n.m('history.education.label.diploma.bachelor')}
                   value="Bachelor"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-master"
                   className="diploma-master"
                   label={i18n.m('history.education.label.diploma.master')}
                   value="Master"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-doctorate"
                   className="diploma-doctorate"
                   label={i18n.m('history.education.label.diploma.doctorate')}
                   value="Doctorate"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-professional"
                   className="diploma-professional"
                   label={i18n.m('history.education.label.diploma.professional')}
                   value="Professional"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-other"
                   className="diploma-other"
                   label={i18n.m('history.education.label.diploma.other')}
                   value="Other"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
          </RadioGroup>
          <Show when={(this.props.Diploma || {}).value === 'Other'}>
            <Field title={i18n.t('history.education.label.diploma.other')}
                   titleSize="label"
                   adjustFor="text"
                   scrollIntoView={this.props.scrollIntoView}>
              <Text name="DiplomaOther"
                    {...this.props.DiplomaOther}
                    className="other"
                    maxlength="100"
                    onUpdate={this.updateDiplomaOther}
                    onError={this.props.onError}
                    required={this.props.required}
                    />
            </Field>
          </Show>
        </Field>

        <Field title={i18n.t('history.education.heading.date')}
               help="history.education.help.date"
               adjustFor="label"
               shrink={true}
               scrollIntoView={this.props.scrollIntoView}>
          <DateControl name="Date"
                       {...this.props.Date}
                       className="date-awarded"
                       hideDay={true}
                       onUpdate={this.updateDate}
                       onError={this.props.onError}
                       required={this.props.required}
                       />
        </Field>
      </div>
    )
  }
}

DiplomaItem.defaultProps = {
  Diploma: {},
  DiplomaOther: {},
  Date: {},
  onUpdate: (queue) => {},
  onError: (value, arr) => { return arr }
}
