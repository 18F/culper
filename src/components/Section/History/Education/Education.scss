.history {
  .school-name {
    width: 92%;
    display: inline-block;
  }

  .type label,
  .diploma label {
    width: 25rem;
    height: 10rem;
    overflow: hidden;
    line-height: normal;
    display: inline-block;

    span {
      display: inline-block;
      padding: 12% 0;
    }
  }
}
