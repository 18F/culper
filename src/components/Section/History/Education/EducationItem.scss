.history {
  .type,
  .diploma {
    label {
      width: 25rem !important;
    }
  }

  .type-college,
  .type-vocational,
  .type-correspondence,
  .diploma-professional {
    label > span > div {
      top: 25%;
    }
  }
}
