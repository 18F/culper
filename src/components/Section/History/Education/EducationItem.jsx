import React from 'react'
import { i18n } from '../../../../config'
import { nameIsEmpty } from '../../../../validators'
import { ValidationElement, BranchCollection, DateRange, Text,
         RadioGroup, Radio, Field, Location, NotApplicable,
         Name, Telephone, Show, Email } from '../../../Form'
import { DiplomaItem } from './Diploma'
import { today, daysAgo } from '../dateranges'

// We need to determine how far back 3 years ago was
const threeYearsAgo = daysAgo(today, 365 * 3)
const withinThreeYears = (from, to) => {
  return (from && from >= threeYearsAgo) || (to && to >= threeYearsAgo)
}

/**
 * Education item in a collection
 *
 * This was broken apart so it could manage minor local state in determining
 * when particular portions of this should be rendered.
 */
export default class EducationItem extends ValidationElement {
  constructor (props) {
    super(props)

    this.update = this.update.bind(this)
    this.updateDates = this.updateDates.bind(this)
    this.updateType = this.updateType.bind(this)
    this.updateName = this.updateName.bind(this)
    this.updateAddress = this.updateAddress.bind(this)
    this.updateComments = this.updateComments.bind(this)
    this.updateReferenceName = this.updateReferenceName.bind(this)
    this.updateReferenceNameNotApplicable = this.updateReferenceNameNotApplicable.bind(this)
    this.updateReferencePhone = this.updateReferencePhone.bind(this)
    this.updateReferenceEmail = this.updateReferenceEmail.bind(this)
    this.updateReferenceEmailNotApplicable = this.updateReferenceEmailNotApplicable.bind(this)
    this.updateReferenceAddress = this.updateReferenceAddress.bind(this)
    this.updateDiplomas = this.updateDiplomas.bind(this)
  }

  update (queue) {
    this.props.onUpdate({
      Dates: this.props.Dates,
      Type: this.props.Type,
      Name: this.props.Name,
      Address: this.props.Address,
      Comments: this.props.Comments,
      ReferenceName: this.props.ReferenceName,
      ReferenceNameNotApplicable: this.props.ReferenceNameNotApplicable,
      ReferencePhone: this.props.ReferencePhone,
      ReferenceEmail: this.props.ReferenceEmail,
      ReferenceEmailNotApplicable: this.props.ReferenceEmailNotApplicable,
      ReferenceAddress: this.props.ReferenceAddress,
      Diplomas: this.props.Diplomas,
      ...queue
    })
  }

  updateDates (values) {
    this.update({
      Dates: values
    })
  }

  updateType (values) {
    this.update({
      Type: values
    })
  }

  updateName (values) {
    this.update({
      Name: values
    })
  }

  updateAddress (values) {
    this.update({
      Address: values
    })
  }

  updateComments (values) {
    this.update({
      Comments: values
    })
  }

  updateReferenceName (values) {
    this.update({
      ReferenceName: values
    })
  }

  updateReferenceNameNotApplicable (values) {
    this.update({
      ReferenceNameNotApplicable: values
    })
  }

  updateReferencePhone (values) {
    this.update({
      ReferencePhone: values
    })
  }

  updateReferenceEmail (values) {
    this.update({
      ReferenceEmail: values
    })
  }

  updateReferenceEmailNotApplicable (values) {
    this.update({
      ReferenceEmailNotApplicable: values
    })
  }

  updateReferenceAddress (values) {
    this.update({
      ReferenceAddress: values
    })
  }

  updateDiplomas (values) {
    this.update({
      Diplomas: values
    })
  }

  diplomaSummary (item, index) {
    const unk = i18n.t('history.education.collection.diploma.summary.unknown')
    const diploma = item.Diploma || {}
    const d = diploma.Date || {}
    const text = (diploma.Diploma === 'Other' ? diploma.DiplomaOther : diploma.Diploma) || unk
    const dd = d.date
          ? `${d.month}/${d.year}`
          : unk

    return (
      <span>
        <span className="index">
          {i18n.t('history.education.collection.diploma.summary.item')} {index + 1}:
        </span>
        <span className=""><strong>{ text }</strong></span>
        <span className="dates"><strong>{ dd }</strong></span>
      </span>
    )
  }

  render () {
    // Certain elements are present if the date range of the attendance was
    // within the last 3 years.
    const dates = this.props.Dates || {}
    const from = (dates.from || {}).date
    const to = (dates.to || {}).date

    return (
      <div className="education">
        <div className="content">
          <Field title={i18n.t('history.education.heading.name')}
                 adjustFor="labels"
                 scrollIntoView={this.props.scrollIntoView}>
            <Text name="Name"
                  {...this.props.Name}
                  label={i18n.t('history.education.label.name')}
                  className="school-name"
                  maxlength="100"
                  onUpdate={this.updateName}
                  onError={this.props.onError}
                  required={this.props.required}
                  />
          </Field>

          <Field title={i18n.t('history.education.heading.dates')}
                 help="history.education.help.dates"
                 adjustFor="daterange"
                 shrink={true}
                 scrollIntoView={this.props.scrollIntoView}>
            <label className="info-label">{i18n.t('history.education.label.dates')}</label>
            <DateRange name="Dates"
                       {...this.props.Dates}
                       label={i18n.t('history.education.label.dates')}
                       onUpdate={this.updateDates}
                       onError={this.props.onError}
                       required={this.props.required}
                       />
          </Field>

          <Field title={i18n.t('history.education.heading.address')}
                 optional={true}
                 help="history.education.help.address"
                 comments={true}
                 commentsName="Comments"
                 commentsValue={this.props.Comments}
                 onUpdate={this.updateComments}
                 onError={this.props.onError}
                 adjustFor="address"
                 shrink={true}
                 scrollIntoView={this.props.scrollIntoView}>
            <Location name="Address"
                      {...this.props.Address}
                      label={i18n.t('history.education.label.address')}
                      layout={Location.ADDRESS}
                      dispatch={this.props.dispatch}
                      addressBooks={this.props.addressBooks}
                      addressBook="Education"
                      geocode={true}
                      onUpdate={this.updateAddress}
                      onError={this.props.onError}
                      required={this.props.required}
                      />
          </Field>

          <Field title={i18n.t('history.education.heading.type')}
                 adjustFor="big-buttons"
                 shrink={true}
                 scrollIntoView={this.props.scrollIntoView}>
            <RadioGroup className="type option-list"
                        required={this.props.required}
                        onError={this.props.onError}
                        selectedValue={(this.props.Type || {}).value}>
              <Radio name="type-highschool"
                     className="type-highschool"
                     label={i18n.m('history.education.label.type.highschool')}
                     value="High School"
                     onUpdate={this.updateType}
                     onError={this.props.onError}
                     />
              <Radio name="type-college"
                     className="type-college"
                     label={i18n.m('history.education.label.type.college')}
                     value="College"
                     onUpdate={this.updateType}
                     onError={this.props.onError}
                     />
              <Radio name="type-vocational"
                     className="type-vocational"
                     label={i18n.m('history.education.label.type.vocational')}
                     value="Vocational"
                     onUpdate={this.updateType}
                     onError={this.props.onError}
                     />
              <Radio name="type-correspondence"
                     className="type-correspondence"
                     label={i18n.m('history.education.label.type.correspondence')}
                     value="Correspondence"
                     onUpdate={this.updateType}
                     onError={this.props.onError}
                     />
            </RadioGroup>
          </Field>

          <Show when={withinThreeYears(from, to)}>
            <div className="reference">
              <Field title={i18n.t('history.education.heading.reference')}
                     titleSize="h2"
                     className="no-margin-bottom">
                {i18n.m('history.education.para.reference')}
              </Field>

              <Field title={i18n.t('reference.heading.name')}
                     titleSize="h3"
                     optional={true}>
                <NotApplicable name="ReferenceNameNotApplicable"
                               {...this.props.ReferenceNameNotApplicable}
                               label={i18n.t('reference.label.idk')}
                               or={i18n.m('reference.para.or')}
                               onUpdate={this.updateReferenceNameNotApplicable}>
                  <Name name="ReferenceName"
                        prefix={'name'}
                        className="reference-name"
                        {...this.props.ReferenceName}
                        onUpdate={this.updateReferenceName}
                        onError={this.props.onError}
                        required={this.props.required}
                        />
                </NotApplicable>
              </Field>

              <Show when={!nameIsEmpty(this.props.ReferenceName)}>
                <Field title={i18n.t('reference.heading.correspondence')}
                       titleSize="h2"
                       optional={true}
                       className="no-margin-bottom">
                  {i18n.m('reference.para.correspondence')}
                </Field>

                <Field title={i18n.t('reference.heading.phone.default')}
                       className="override-required"
                       help={'reference.help.phone'}
                       adjustFor="telephone"
                       scrollIntoView={this.props.scrollIntoView}>
                  <Telephone name="ReferencePhone"
                             className="reference-phone"
                             {...this.props.ReferencePhone}
                             onUpdate={this.updateReferencePhone}
                             onError={this.props.onError}
                             required={this.props.required}
                             />
                </Field>

                <Field title={i18n.t('reference.heading.email')}
                       help={'reference.help.email'}
                       adjustFor="label">
                  <NotApplicable name="ReferenceEmailNotApplicable"
                                 {...this.props.ReferenceEmailNotApplicable}
                                 label={i18n.t('reference.label.idk')}
                                 or={i18n.m('reference.para.or')}
                                 onUpdate={this.updateReferenceEmailNotApplicable}>
                    <Email name="ReferenceEmail"
                           {...this.props.ReferenceEmail}
                           className="reference-email"
                           label={i18n.t('reference.label.email')}
                           onUpdate={this.updateReferenceEmail}
                           onError={this.props.onError}
                           />
                  </NotApplicable>
                </Field>

                <Field title={i18n.t('reference.heading.address')}
                       optional={true}
                       help={'reference.help.address'}
                       adjustFor="address">
                  <p>{i18n.t('reference.para.address')}</p>
                  <Location name="ReferenceAddress"
                            className="reference-address"
                            {...this.props.ReferenceAddress}
                            label={i18n.t('reference.label.address')}
                            layout={Location.ADDRESS}
                            geocode={true}
                            addressBooks={this.props.addressBooks}
                            addressBook="Reference"
                            dispatch={this.props.dispatch}
                            onUpdate={this.updateReferenceAddress}
                            onError={this.props.onError}
                            />
                </Field>
              </Show>
            </div>
          </Show>

          <BranchCollection label={i18n.t('history.education.heading.degree')}
                            appendLabel={i18n.t('history.education.heading.degreeTail')}
                            className="receive-degree"
                            {...this.props.Diplomas}
                            onUpdate={this.updateDiplomas}
                            onError={this.props.onError}
                            required={this.props.required}
                            scrollIntoView={this.props.scrollIntoView}>
            <DiplomaItem name="Item" bind={true} required={this.props.required} scrollIntoView={this.props.scrollIntoView} />
          </BranchCollection>
        </div>
      </div>
    )
  }
}

EducationItem.defaultProps = {
  Dates: {},
  Type: {},
  Name: {},
  Address: {},
  Comments: {},
  ReferenceName: {},
  ReferenceNameNotApplicable: {},
  ReferencePhone: {},
  ReferenceEmail: {},
  ReferenceEmailNotApplicable: {},
  ReferenceAddress: {},
  Diplomas: { items: [] },
  onUpdate: (queue) => {},
  onError: (value, arr) => { return arr }
}
