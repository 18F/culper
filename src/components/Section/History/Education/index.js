import Education from './Education'
export default Education
