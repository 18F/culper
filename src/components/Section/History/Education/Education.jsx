import React from 'react'
import { i18n } from '../../../../config'
import schema from '../../../../schema'
import validate, { EducationItemValidator } from '../../../../validators'
import SubsectionElement from '../../SubsectionElement'
import { Accordion } from '../../../Form'
import { openState } from '../../../Form/Accordion/Accordion'
import { EducationCustomSummary } from '../summaries'
import EducationItem from './EducationItem'

const byline = (item, index, initial, translation, required, validator) => {
  switch (true) {
    // If item is required and not currently opened and is not valid, show message
  case required && !item.open && !validator(item.Item):
  case !item.open && !initial && item.Item && !validator(item.Item):
    return (<div className={`byline ${openState(item, initial)} fade in`.trim()}>
            <div className="incomplete">{i18n.m(translation)}</div>
            </div>
           )
  default:
    return null
  }
}

export default class Education extends SubsectionElement {
  constructor (props) {
    super(props)

    this.customEducationByline = this.customEducationByline.bind(this)
  }

  customEducationByline (item, index, initial) {
    return byline(item, index, this.props.overrideInitial(initial), 'history.education.collection.school.summary.incomplete', this.props.required, (item) => {
      return new EducationItemValidator(item).isValid()
    })
  }

  render () {
    return (
      <div className="education">
        <Accordion scrollToTop={this.props.scrollToTop}
                   defaultState={this.props.defaultState}
                   {...this.props.List}
                   sort={this.props.sort}
                   realtime={this.props.realtime}
                   onUpdate={this.props.onUpdate}
                   onError={this.handleError}
                   caption={this.props.caption}
                   byline={this.customEducationByline}
                   customSummary={EducationCustomSummary}
                   description={i18n.t('history.education.collection.school.summary.title')}
                   appendTitle={i18n.t('history.education.collection.school.appendTitle')}
                   appendLabel={i18n.t('history.education.collection.school.append')}
                   required={this.props.required}
                   scrollIntoView={this.props.scrollIntoView}>
          <EducationItem name="Item"
                         bind={true}
                         required={this.props.required}
                         scrollIntoView={this.props.scrollIntoView}
                         addressBooks={this.props.addressBooks}
                         dispatch={this.props.dispatch}
                         />
        </Accordion>
      </div>
    )
  }
}

Education.defaultProps = {
  List: Accordion.defaultList,
  scrollToTop: '',
  defaultState: true,
  realtime: false,
  sort: null,
  totalYears: 10,
  overrideInitial: (initial) => { return initial },
  caption: null,
  onUpdate: () => {},
  onError: (value, arr) => { return arr },
  section: 'history',
  subsection: 'education',
  dispatch: () => {},
  validator: (state, props) => {
    return validate(schema('history.education', props))
  }
}
