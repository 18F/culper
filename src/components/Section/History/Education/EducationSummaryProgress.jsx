import React from 'react'
import PropTypes from 'prop-types'

import { i18n } from 'config'

import SummaryCounter from 'components/Section/History/SummaryCounter'
import { totalYears } from 'components/Section/History/helpers'

import { EducationItemValidator } from 'validators'

const schoolRangesList = (items) => {
  const dates = []

  items.forEach((i) => {
    if (!i.Item || !i.Item.Dates || !i.Item.Dates.to || !i.Item.Dates.from) {
      return
    }

    if (new EducationItemValidator(i.Item).isValid()) {
      dates.push(i.Item.Dates)
    }
  })

  return dates
}

const diplomasRangesList = (items) => {
  const dates = []

  items.forEach((i) => {
    if (!i.Item) { return }

    if (!new EducationItemValidator(i.Item).isValid()) {
      return
    }

    if (i.Item.Diplomas.items) {
      i.Item.Diplomas.items.forEach((d) => {
        if (!d.Item || !d.Item.Date) { return }

        dates.push(d.Item.Date)
      })
    }
  })

  return dates
}

const EducationSummaryProgress = (props) => {
  const { Education, Birthdate, years } = props

  let schoolDates = []
  if (Education && Education.List && Education.List.items) {
    schoolDates = Education.List.items
  }

  const getSchoolDates = () => schoolRangesList(schoolDates)
  const getDiplomaDates = () => diplomasRangesList(schoolDates)

  return (
    <SummaryCounter
      className="education"
      title={i18n.t('history.education.summary.title')}
      schools={getSchoolDates}
      diplomas={getDiplomaDates}
      schoolsLabel={i18n.t('history.education.summary.schools')}
      diplomasLabel={i18n.t('history.education.summary.diplomas')}
      total={totalYears(Birthdate, years)}
    />
  )
}

/* eslint react/forbid-prop-types: 0 */
EducationSummaryProgress.propTypes = {
  Education: PropTypes.object,
  Birthdate: PropTypes.any,
  years: PropTypes.number,
}

EducationSummaryProgress.defaultProps = {
  Education: null,
  Birthdate: null,
  years: 10,
}

export default EducationSummaryProgress
