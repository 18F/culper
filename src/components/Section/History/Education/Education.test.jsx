import React from 'react'
import { shallow, mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { Education } from './Education'

describe('The Education component', () => {
  it('renders without errors', () => {
    const component = shallow(<Education />)

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })

  it('renders an education item', () => {
    const component = shallow(<Education />)

    expect(component.find('.education').length).toEqual(1)
  })

  const mockStore = configureMockStore()

  const mountComponentWithStore = (props = {}, defaultState = {}) => {
    const store = mockStore({ ...defaultState })

    return mount(
      <Provider store={store}>
        <Education {...props} />
      </Provider>
    )
  }

  describe('with one item', () => {
    const testProps = {
      List: {
        items: [{ Item: {} }],
      },
    }

    const component = mountComponentWithStore(testProps)

    it('renders two education items', () => {
      expect(component.find('.education').length).toEqual(2)
    })
  })
})
