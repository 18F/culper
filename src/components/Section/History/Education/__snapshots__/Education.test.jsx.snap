// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Education component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="section-content education"
    data-section="history"
    data-subsection="education"
  >
    <Accordion
      appendClass=""
      appendHelp={null}
      appendLabel="Add another school"
      appendMessage={null}
      appendTitle="Do you have additional education (include education within the last 10 years, as well as degrees or diplomas more than 10 years ago)?"
      branch={Object {}}
      byline={[Function]}
      caption={null}
      className=""
      closeLabel="Close"
      customDetails={[Function]}
      customSummary={[Function]}
      defaultState={true}
      description="Summary of education"
      incomplete="There is a problem with this item, please open it and fix the issue(s)."
      initial={true}
      inject={[Function]}
      items={Array []}
      minimum={1}
      onError={[Function]}
      onUpdate={[Function]}
      openLabel="Open"
      realtime={false}
      removeLabel="Remove"
      scrollIntoView={false}
      scrollToBottom=""
      scrollToTop=""
      skipWarning={false}
      sort={null}
      summary={[Function]}
      timeout={500}
      transformer={[Function]}
      validator={[Function]}
    >
      <EducationItem
        Address={Object {}}
        Comments={Object {}}
        Dates={Object {}}
        Diplomas={
          Object {
            "items": Array [],
          }
        }
        Name={Object {}}
        ReferenceAddress={Object {}}
        ReferenceEmail={Object {}}
        ReferenceEmailNotApplicable={Object {}}
        ReferenceName={Object {}}
        ReferenceNameNotApplicable={Object {}}
        ReferencePhone={Object {}}
        Type={Object {}}
        bind={true}
        dispatch={[Function]}
        name="Item"
        onError={[Function]}
        onUpdate={[Function]}
        scrollIntoView={false}
      />
    </Accordion>
  </div>,
  "nodes": Array [
    <div
      className="section-content education"
      data-section="history"
      data-subsection="education"
    >
      <Accordion
        appendClass=""
        appendHelp={null}
        appendLabel="Add another school"
        appendMessage={null}
        appendTitle="Do you have additional education (include education within the last 10 years, as well as degrees or diplomas more than 10 years ago)?"
        branch={Object {}}
        byline={[Function]}
        caption={null}
        className=""
        closeLabel="Close"
        customDetails={[Function]}
        customSummary={[Function]}
        defaultState={true}
        description="Summary of education"
        incomplete="There is a problem with this item, please open it and fix the issue(s)."
        initial={true}
        inject={[Function]}
        items={Array []}
        minimum={1}
        onError={[Function]}
        onUpdate={[Function]}
        openLabel="Open"
        realtime={false}
        removeLabel="Remove"
        scrollIntoView={false}
        scrollToBottom=""
        scrollToTop=""
        skipWarning={false}
        sort={null}
        summary={[Function]}
        timeout={500}
        transformer={[Function]}
        validator={[Function]}
      >
        <EducationItem
          Address={Object {}}
          Comments={Object {}}
          Dates={Object {}}
          Diplomas={
            Object {
              "items": Array [],
            }
          }
          Name={Object {}}
          ReferenceAddress={Object {}}
          ReferenceEmail={Object {}}
          ReferenceEmailNotApplicable={Object {}}
          ReferenceName={Object {}}
          ReferenceNameNotApplicable={Object {}}
          ReferencePhone={Object {}}
          Type={Object {}}
          bind={true}
          dispatch={[Function]}
          name="Item"
          onError={[Function]}
          onUpdate={[Function]}
          scrollIntoView={false}
        />
      </Accordion>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Education
        List={
          Object {
            "branch": Object {},
            "items": Array [],
          }
        }
        caption={null}
        defaultState={true}
        dispatch={[Function]}
        onError={[Function]}
        onUpdate={[Function]}
        overrideInitial={false}
        realtime={false}
        scrollIntoView={false}
        scrollToTop=""
        section="history"
        sort={null}
        subsection="education"
        totalYears={10}
        validator={[Function]}
      />,
      "_debugID": 381,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Education {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "customEducationByline": [Function],
        "handleCompletion": [Function],
        "handleError": [Function],
        "props": Object {
          "List": Object {
            "branch": Object {},
            "items": Array [],
          },
          "caption": null,
          "defaultState": true,
          "dispatch": [Function],
          "onError": [Function],
          "onUpdate": [Function],
          "overrideInitial": false,
          "realtime": false,
          "scrollIntoView": false,
          "scrollToTop": "",
          "section": "history",
          "sort": null,
          "subsection": "education",
          "totalYears": 10,
          "validator": [Function],
        },
        "refs": Object {},
        "section": "history",
        "state": null,
        "store": "History",
        "storeKey": "Education",
        "subsection": "education",
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="section-content education"
          data-section="history"
          data-subsection="education"
        >
          <Accordion
            appendClass=""
            appendHelp={null}
            appendLabel="Add another school"
            appendMessage={null}
            appendTitle="Do you have additional education (include education within the last 10 years, as well as degrees or diplomas more than 10 years ago)?"
            branch={Object {}}
            byline={[Function]}
            caption={null}
            className=""
            closeLabel="Close"
            customDetails={[Function]}
            customSummary={[Function]}
            defaultState={true}
            description="Summary of education"
            incomplete="There is a problem with this item, please open it and fix the issue(s)."
            initial={true}
            inject={[Function]}
            items={Array []}
            minimum={1}
            onError={[Function]}
            onUpdate={[Function]}
            openLabel="Open"
            realtime={false}
            removeLabel="Remove"
            scrollIntoView={false}
            scrollToBottom=""
            scrollToTop=""
            skipWarning={false}
            sort={null}
            summary={[Function]}
            timeout={500}
            transformer={[Function]}
            validator={[Function]}
          >
            <EducationItem
              Address={Object {}}
              Comments={Object {}}
              Dates={Object {}}
              Diplomas={
                Object {
                  "items": Array [],
                }
              }
              Name={Object {}}
              ReferenceAddress={Object {}}
              ReferenceEmail={Object {}}
              ReferenceEmailNotApplicable={Object {}}
              ReferenceName={Object {}}
              ReferenceNameNotApplicable={Object {}}
              ReferencePhone={Object {}}
              Type={Object {}}
              bind={true}
              dispatch={[Function]}
              name="Item"
              onError={[Function]}
              onUpdate={[Function]}
              scrollIntoView={false}
            />
          </Accordion>
        </div>,
        "_debugID": 382,
        "_renderedOutput": <div
          className="section-content education"
          data-section="history"
          data-subsection="education"
        >
          <Accordion
            appendClass=""
            appendHelp={null}
            appendLabel="Add another school"
            appendMessage={null}
            appendTitle="Do you have additional education (include education within the last 10 years, as well as degrees or diplomas more than 10 years ago)?"
            branch={Object {}}
            byline={[Function]}
            caption={null}
            className=""
            closeLabel="Close"
            customDetails={[Function]}
            customSummary={[Function]}
            defaultState={true}
            description="Summary of education"
            incomplete="There is a problem with this item, please open it and fix the issue(s)."
            initial={true}
            inject={[Function]}
            items={Array []}
            minimum={1}
            onError={[Function]}
            onUpdate={[Function]}
            openLabel="Open"
            realtime={false}
            removeLabel="Remove"
            scrollIntoView={false}
            scrollToBottom=""
            scrollToTop=""
            skipWarning={false}
            sort={null}
            summary={[Function]}
            timeout={500}
            transformer={[Function]}
            validator={[Function]}
          >
            <EducationItem
              Address={Object {}}
              Comments={Object {}}
              Dates={Object {}}
              Diplomas={
                Object {
                  "items": Array [],
                }
              }
              Name={Object {}}
              ReferenceAddress={Object {}}
              ReferenceEmail={Object {}}
              ReferenceEmailNotApplicable={Object {}}
              ReferenceName={Object {}}
              ReferenceNameNotApplicable={Object {}}
              ReferencePhone={Object {}}
              Type={Object {}}
              bind={true}
              dispatch={[Function]}
              name="Item"
              onError={[Function]}
              onUpdate={[Function]}
              scrollIntoView={false}
            />
          </Accordion>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Education
    List={
      Object {
        "branch": Object {},
        "items": Array [],
      }
    }
    caption={null}
    defaultState={true}
    dispatch={[Function]}
    onError={[Function]}
    onUpdate={[Function]}
    overrideInitial={false}
    realtime={false}
    scrollIntoView={false}
    scrollToTop=""
    section="history"
    sort={null}
    subsection="education"
    totalYears={10}
    validator={[Function]}
  />,
}
`;
